### PR TITLE
Integrate Message into Transaction

### PR DESCRIPTION
**Pull request by garious**
#### Problem

We construct a Transaction from a Message, but Transaction doesn't store it. Instead, it copies over all the same information. Then to sign that information, we reconstruct the original message, serialize it and sign that.  It works and but anytime you want to make a change to Message, you need to make the same change in Transaction. It's a maintenance burden.

#### Summary of Changes

Integrate Message into Transaction

Towards: #3312

### CODE REVIEW

File: core/src/sigverify.rs [hunk 1]
```diff
@@ -128,7 +128,7 @@ pub fn get_packet_offsets(packet: &Packet, current_offset: u32) -> (u32, u32, u3
 
     let sig_start = current_offset as usize + sig_size;
     let msg_start = current_offset as usize + msg_start_offset;
-    let pubkey_start = msg_start + pubkey_size;
+    let pubkey_start = msg_start + 1 + pubkey_size;
```
Comment (rob-solana) [inline core/src/sigverify.rs]: what's this magic #?

**Comment by rob-solana** [comment]
what's the definition of "Message" now?

File: sdk/src/message.rs [hunk 2]
```diff
@@ -70,34 +70,67 @@ fn get_program_ids(instructions: &[Instruction]) -> Vec<Pubkey> {
...
+        Self {
+            num_signatures,
+            account_keys,
+            recent_blockhash,
+            fee,
+            program_ids,
+            instructions,
+        }
+    }
+
```
Comment (CriesofCarrots) [inline sdk/src/message.rs]: The following comment has a typo, but I think it's also incorrect now. It doesn't return a transaction.

File: sdk/src/transaction.rs [hunk 3]
```diff
@@ -374,14 +351,15 @@ mod tests {
         let expected_transaction_size = 1
             + (tx.signatures.len() * size_of::<Signature>())
             + 1
-            + (tx.account_keys.len() * size_of::<Pubkey>())
+            + 1
```
Comment (CriesofCarrots) [inline sdk/src/transaction.rs]: Same question as Rob's about this magic number. I assume it is the num_signatures length, but some comments on these integers might make it more clear, and make upkeep easier.

File: core/src/sigverify.rs (hunk 1 above)
Comment (garious) [reply core/src/sigverify.rs]: `size_of(Message::num_signatures)` . I'll add a constant.

**Comment by garious** [comment]
@rob-solana, let's discuss a definition for "message" over in the terminology channel.  In this PR, it's the deserialized information that requires signatures from the first `num_signatures` of `account_keys`.  It's the thing that a client should serialize and send to a drone or vote-signer, which could then deserialize and inspect before signing that data and returning the signature.

File: sdk/src/message.rs (hunk 2 above)
Comment (garious) [reply sdk/src/message.rs]: Thanks, that was copied from Transaction. I deleted it.

File: core/src/sigverify.rs (hunk 1 above)
Comment (garious) [reply core/src/sigverify.rs]: done

File: sdk/src/transaction.rs (hunk 3 above)
Comment (garious) [reply sdk/src/transaction.rs]: done

**Pull request merged**